src/main/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheck.java

```diff
@@ -91,14 +91,9 @@ protected void processFiltered(File file, FileText fileText) {
      *         one of 'crlf', 'lf', 'cr', 'lf_cr_crlf' or 'system'
      */
     public void setLineSeparator(String lineSeparatorParam) {
-        try {
-            lineSeparator =
-                Enum.valueOf(LineSeparatorOption.class, lineSeparatorParam.trim()
-                    .toUpperCase(Locale.ENGLISH));
-        }
-        catch (IllegalArgumentException iae) {
-            throw new IllegalArgumentException("unable to parse " + lineSeparatorParam, iae);
-        }
+        lineSeparator =
+            Enum.valueOf(LineSeparatorOption.class, lineSeparatorParam.trim()
+                .toUpperCase(Locale.ENGLISH));
```

Comment:
This is basically amounts to a no change as no changing tests show.
The exception message generated by `AutomaticBean` when a conversion fails specifically says what value was trying to be assigned.

The trace will change from:
````
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck - Cannot set property 'lineSeparator' to 'ct' in module com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:463)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:198)
	at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.createChecker(AbstractModuleTestSupport.java:144)
	at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.createChecker(AbstractModuleTestSupport.java:118)
	at com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheckTest.testSetLineSeparatorFailure(NewlineAtEndOfFileCheckTest.java:149)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:538)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:760)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:460)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:206)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Cannot set property 'lineSeparator' to 'ct' in module com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.tryCopyProperty(AutomaticBean.java:238)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:191)
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:458)
	... 27 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.commons.beanutils.PropertyUtilsBean.invokeMethod(PropertyUtilsBean.java:2127)
	at org.apache.commons.beanutils.PropertyUtilsBean.setSimpleProperty(PropertyUtilsBean.java:2108)
	at org.apache.commons.beanutils.BeanUtilsBean.copyProperty(BeanUtilsBean.java:437)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.tryCopyProperty(AutomaticBean.java:228)
	... 29 more
Caused by: java.lang.IllegalArgumentException: unable to parse ct
	at com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck.setLineSeparator(NewlineAtEndOfFileCheck.java:100)
	... 37 more
Caused by: java.lang.IllegalArgumentException: No enum constant com.puppycrawl.tools.checkstyle.checks.LineSeparatorOption.CT
	at java.lang.Enum.valueOf(Enum.java:238)
	at com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck.setLineSeparator(NewlineAtEndOfFileCheck.java:96)
	... 37 more
````

to

````
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck - Cannot set property 'lineSeparator' to 'ct' in module com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:463)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:198)
	at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.createChecker(AbstractModuleTestSupport.java:144)
	at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.createChecker(AbstractModuleTestSupport.java:118)
	at com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheckTest.testSetLineSeparatorFailure(NewlineAtEndOfFileCheckTest.java:149)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:538)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:760)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:460)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:206)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Cannot set property 'lineSeparator' to 'ct' in module com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.tryCopyProperty(AutomaticBean.java:238)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:191)
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:458)
	... 27 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.commons.beanutils.PropertyUtilsBean.invokeMethod(PropertyUtilsBean.java:2127)
	at org.apache.commons.beanutils.PropertyUtilsBean.setSimpleProperty(PropertyUtilsBean.java:2108)
	at org.apache.commons.beanutils.BeanUtilsBean.copyProperty(BeanUtilsBean.java:437)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.tryCopyProperty(AutomaticBean.java:228)
	... 29 more
Caused by: java.lang.IllegalArgumentException: No enum constant com.puppycrawl.tools.checkstyle.checks.LineSeparatorOption.CT
	at java.lang.Enum.valueOf(Enum.java:238)
	at com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck.setLineSeparator(NewlineAtEndOfFileCheck.java:95)
	... 37 more
````

No regression can be given for this type of change.

Reply:
If only stacktrace is changing , it ok completely

Reply:
@romani That is correct. Only stack trace.
It basically amounts to this section being removed:
````
Caused by: java.lang.IllegalArgumentException: unable to parse ct
	at com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck.setLineSeparator(NewlineAtEndOfFileCheck.java:100)
	... 37 more
````

We still have
`IllegalArgumentException: No enum constant com.puppycrawl.tools.checkstyle.checks.LineSeparatorOption.CT`
and
`Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Cannot set property 'lineSeparator' to 'ct' in module com.puppycrawl.tools.checkstyle.checks.NewlineAtEndOfFileCheck`.